plugins/modules/ndo_l3out_interface_group_policy.py

```diff
@@ -0,0 +1,562 @@
+            existing_interface_groups,
+            [KVPair("name", name)],
+        )
+        if match:
+            set_names_for_references(mso, match.details)
+            mso.existing = mso.previous = copy.deepcopy(match.details)
+    else:
+        mso.existing = mso.previous = [set_names_for_references(mso, interface_group) for interface_group in existing_interface_groups]
+
+    if state != "query":
+        interface_group_policy_path = "/l3outTemplate/l3outs/{0}/interfaceGroups/{1}".format(l3out_object.index, match.index if match else "-")
+
+    ops = []
+
+    attributes_dict = {"bfd": bfd, "bfdMultiHop": bfd_multi_hop, "ospf": ospf}
+    empty_interface_routing_policy = False
+    empty_custom_qos_policy = False
+
+    if state == "present":
+        check_authentication_requirements(mso, attributes_dict)
+        mso_values = dict(
+            name=name,
+            description=description,
+            interfaceRoutingPolicyRef=interface_routing_policy_uuid,
+            qosPriority=qos_priority,
+            qosRef=custom_qos_policy_uuid,
+        )
+        if interface_routing_policy:
+            empty_interface_routing_policy = check_if_all_elements_are_none(list(interface_routing_policy.values()))
+            if not empty_interface_routing_policy:
+                mso_values["interfaceRoutingPolicyRef"] = mso_template.get_l3out_interface_routing_policy_uuid(
+                    mso, interface_routing_policy.get("name"), interface_routing_policy.get("template")
+                )
+        if custom_qos_policy:
+            empty_custom_qos_policy = check_if_all_elements_are_none(list(custom_qos_policy.values()))
+            if not empty_custom_qos_policy:
+                mso_values["qosRef"] = mso_template.get_custom_qos_policy_uuid(mso, custom_qos_policy.get("name"), custom_qos_policy.get("template"))
+
+        if match:
+            mso_values_remove = list()
+
+            if (interface_routing_policy_uuid == "" or empty_interface_routing_policy) and match.details.get("interfaceRoutingPolicyRef"):
+                mso_values_remove.append("interfaceRoutingPolicyRef")
+            if (custom_qos_policy_uuid == "" or empty_custom_qos_policy) and match.details.get("qosRef"):
+                mso_values_remove.append("qosRef")
+
+            for attribute_name, attribute_dict in attributes_dict.items():
+                if attribute_dict is not None:
+                    attribute_is_empty = check_if_all_elements_are_none(list(attribute_dict.values()))
+
+                    if match.details.get(attribute_name, {}).get("key", {}).get("ref"):
+                        match.details[attribute_name]["key"].pop("ref", None)
+                        mso.previous[attribute_name]["key"].pop("ref", None)
+                        mso.existing[attribute_name]["key"].pop("ref", None)
+
+                    if attribute_is_empty and match.details.get(attribute_name):
+                        mso_values_remove.append(attribute_name)
```

Comment:
just so I understand, this is required for the removal of authentication for a specific protocol?

Reply:
This for removal of both the protocol and the authentication of a protocol. Without the protocol there's no authentication required for it. (example -> bfd: {})